dashboard/run_old_data_collection.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+"""
+Script for old dashboard data collection.
+Stops running once finish collecting all the old core hour usage data by the specified collection period
+
+The collected data populates the specified dashboard db.
+"""
+
+import argparse
+import os
+import subprocess
+from typing import Iterable, List, Union
+from datetime import datetime, timedelta
+
+import qcore.constants as const
+from dashboard.run_data_collection import HPCS, USERS, DataCollector
+from dashboard.DashboardDB import DashboardDB
+
+DEFAULT_DAYS_SHIFT = 10
+
+
+def get_old_utc_times(day_shift: int):
+    """Get the hpc utc time based on current real time - day_shift
+       To be used as the start_time of sreport cmd
+    """
+    # 2019-03-25
+    local_date = datetime.today().date()
+    # if day_shift == 1: 2019-03-24 00:00:00
+    local_start_datetime = datetime(
+        local_date.year, local_date.month, local_date.day
+    ) - timedelta(days=day_shift)
+    utc_start_datetime = local_start_datetime - DataCollector.utc_time_gap
+    start_utc_time_string = utc_start_datetime.strftime(DataCollector.utc_time_format)
+    end_utc_time_string = (utc_start_datetime + timedelta(days=1)).strftime(
+        DataCollector.utc_time_format
+    )
+    return local_start_datetime.date(), start_utc_time_string, end_utc_time_string
+
+
+def collect_old_data(
+    dashboard_db: DashboardDB,
+    hpc: const.HPC,
+    login_user: str,
+    users: Iterable[str],
+    day_shift: int,
+):
+    """Collect old core hours usage for a specified users in a previous day"""
+    local_date, start_time, end_time = get_old_utc_times(day_shift)
+    cmd = "sreport -M {} -t Hours cluster AccountUtilizationByUser Users={} start={} end={} -n".format(
+        hpc.value, " ".join(users), start_time, end_time
+    )
+    user_ch_output = (
+        subprocess.check_output(
+            "ssh {}@{} {}".format(login_user, hpc.value, cmd), shell=True, timeout=10
+        )
+        .decode("utf-8")
+        .strip()
+        .split("\n")
+    )
+    if user_ch_output:
+        dashboard_db.update_user_chours(
+            hpc,
+            DataCollector.parse_user_chours_usage(user_ch_output, users, local_date),
+            local_date,
+        )
+
+
+def run_old_collection(dashboard_db, hpcs: Union[List[const.HPC], const.HPC], login_user: str, users: Iterable[str], days_shift: int):
+    """Runs the data collection for a specified period"""
+    # Iterate through the specifid days period
+    hpcs = [hpcs] if type(hpcs) is const.HPC else hpcs
+    for day_shift in range(days_shift + 1):
+        # Collect the data
+        print("{} - Collecting data from HPC {}".format(datetime.now(), hpcs))
+        for hpc in hpcs:
+            collect_old_data(dashboard_db, hpc, login_user, users, day_shift)
+            print("{} - Done".format(datetime.now()))
+
+
+def main(args):
+    hpc = (
+        const.HPC(args.hpc)
```

Comment:
Should be [const.HPC(args.hpc)], so that you're always passing a list into run_old_collection, and don't need line 71?

Reply:
I tried this before, but then the 'hpc' would not work for function "collect_old_data"